docs/configuration/http_filters/cors_filter.rst

```diff
@@ -0,0 +1,60 @@
+.. _config_http_filters_cors:
+
+CORS filter
+====================
+
+This is a filter which handles CORS requests based on route or virtual host settings.
+For the meaning of the headers please refer to the pages below.
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
+https://www.w3.org/TR/cors/
+
+
+.. code-block:: json
+
+  {
+    "type": "both",
+    "name": "cors",
+    "config": {}
+  }
+
+
+Settings
+--------
+
+Settings on a route take precedence over settings on the virtual host.
+Both the route and the virtual host need to set enabled to true.
+
+.. code-block:: json
+
+  {
+    "cors": {
+        "enabled": false,
+        "allow_origin": "",
+        "allow_methods": "",
+        "allow_headers": "",
+        "allow_credentials": false,
+        "expose_headers": "",
+        "max_age": ""
+    }
+  }
+
+enabled
+  *(optional, boolean)* Defaults to false.
+
+allow_origin
```

Comment:
Curious why is this required when the code allows it to be empty?

Reply:
I wanted to get some feedback of what you think should be the required parameters. I think making `enabled` required would make sense too.

True, I didn't think about that much. I should have checked if it's the empty string.